testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -350,47 +502,55 @@ pub fn clean_k8s_cluster(
-        let validator_upgrade_args = [
-            "upgrade",
-            &format!("val{}", i),
-            &format!("{}/diem-validator", helm_repo),
-            "--reuse-values",
-            "--set",
-            &format!("chain.era={}", new_era),
-        ];
-        println!("{:?}", validator_upgrade_args);
-        let validator_upgrade_output = Command::new(HELM_BIN)
-            .stdout(Stdio::inherit())
-            .args(&validator_upgrade_args)
-            .output()
-            .expect("failed to helm upgrade diem");
-        assert!(validator_upgrade_output.status.success());
-    });
+    assert!(
+        testnet_upgrade_output.status.success(),
+        "{}",
+        String::from_utf8(testnet_upgrade_output.stderr).unwrap()
+    );
 
-    let kube_client = rt.block_on(K8sClient::try_default()).unwrap();
+    let rt = Runtime::new().unwrap();
 
-    rt.block_on(async {
+    // wait for genesis to run again, and get the updated validators
+    let mut validators = rt.block_on(async {
+        let mut kube_proxy = Command::new(KUBECTL_BIN).arg("proxy").spawn().unwrap();
         diem_retrier::retry_async(k8s_retry_strategy(), || {
-            let jobs: Api<Job> = Api::namespaced(kube_client.clone(), "default");
             Box::pin(async move {
-                let job_name = format!("diem-testnet-genesis-e{}", new_era);
-                println!("Running get job: {}", &job_name);
-                let genesis_job = jobs.get_status(&job_name).await.unwrap();
-                println!("Status: {:?}", genesis_job.status);
-                let status = genesis_job.status.unwrap();
-                match status.succeeded {
-                    Some(1) => {
-                        println!("Genesis job completed");
-                        Ok(())
-                    }
-                    _ => bail!("Genesis job not completed"),
-                }
+                debug!("Running local kube pod healthcheck on {}", HEALTH_CHECK_URL);
+                reqwest::get(HEALTH_CHECK_URL).await?.text().await?;
+                println!("Local kube pod healthcheck passed");
+                Ok::<(), reqwest::Error>(())
             })
         })
         .await
-    })
+        .expect("Failed kube proxy healthcheck");
+        let config = Config::new(
+            reqwest::Url::parse(HEALTH_CHECK_URL).expect("Failed to parse kubernetes endpoint url"),
+        );
+        let kube_client = K8sClient::try_from(config).unwrap();
+        wait_genesis_job(&kube_client, new_era).await.unwrap();
+        let vals = get_validators(kube_client.clone()).await.unwrap();
+        kube_proxy.kill().unwrap();
+        vals
+    });
+
+    // healthcheck on each of the validators
+    let unhealthy_validators = validators
+        .iter_mut()
+        .filter_map(|(_, val)| {
+            println!("Attempting health check: {}", val.name.clone());
+            match val.health_check() {
+                Ok(_) => Some(val.name.clone()),
```

Comment:
is the logic here adding val name to unhealthy_validators if val.health_check() is Ok()?

Reply:
yep it's mainly for debugging purposes

Reply:
u'd better add an extra struct to hold info for debugging purpose instead of changing logic reversely which does have high risk

Reply:
Updated: I instead did `fmt::Debug` for `K8sNode` to simplify this filter statement.